synapse/handlers/directory.py

```diff
@@ -84,6 +84,13 @@ def get_association(self, room_alias):
             if result:
                 room_id = result.room_id
                 servers = result.servers
+            else:
+                # Query AS to see if it exists
+                as_handler = self.hs.get_handlers().appservice_handler
+                result = yield as_handler.query_room_alias_exists(room_alias)
+                if result:
+                    room_id = result.room_id
+                    servers = result.servers
```

Comment:
Good call, didn't see that when adding this. Will factor out the duplicate logic.